src/main/java/com/marklogic/mule/extension/Operations.java

```diff
@@ -112,66 +113,100 @@ public void writeDocuments(
      */
     @MediaType(value = ANY, strict = false)
     @DisplayName("Search Documents")
-    public List<Result<InputStream, DocumentAttributes>> searchDocuments(
-        @Connection DatabaseClient databaseClient,
+    public PagingProvider<DatabaseClient, Result<InputStream, DocumentAttributes>> searchDocuments(
         @DisplayName("Collection") @Optional(defaultValue = "") @Example("myCollection") String collection,
         @DisplayName("Query") @Text @Optional @Example("searchTerm") String query,
         @DisplayName("Query Type") @Optional QueryType queryType,
         @DisplayName("Query Format") @Optional QueryFormat queryFormat,
         @DisplayName("Metadata Category List") @Optional(defaultValue = "all") @Example("COLLECTIONS,PERMISSIONS") String categories,
         @DisplayName("Max Results") @Optional @Example("10") Integer maxResults,
+        @DisplayName("Page Length") @Optional @Example("10") Integer pageLength,
         @DisplayName("Search Options") @Optional @Example("appSearchOptions") String searchOptions,
         @DisplayName("Directory") @Optional @Example("/customerData") String directory,
         @DisplayName("REST Transform") @Optional String restTransform,
         @DisplayName("REST Transform Parameters") @Optional String restTransformParameters,
         @DisplayName("REST Transform Parameters Delimiter") @Optional @Example(",") String restTransformParametersDelimiter
     ) {
-        DocumentManager<JSONReadHandle, JSONWriteHandle> documentManager = databaseClient.newJSONDocumentManager();
-        if (Utilities.hasText(categories)) {
-            documentManager.setMetadataCategories(buildMetadataCategories(categories));
-        }
-        if (maxResults != null) {
-            documentManager.setPageLength(maxResults);
+        final int finalPageLength;
```

Comment:
I think this is begging for a ternary operator. :)

    final int finalPageLength = pageLength != null ? pageLength : 10; 

Let's default it to 100 as well, which is a pretty standard batch size for us - e.g. MLCP  and DMSDK - and is usually pretty good for performance.